subprojects/jvm-services/src/test/groovy/org/gradle/internal/jvm/inspection/DefaultJvmMetadataDetectorTest.groovy

```diff
@@ -431,8 +464,13 @@ class DefaultJvmMetadataDetectorTest extends Specification {
         def handle = Mock(ExecHandle)
         handle.start() >> handle
         handle.waitForFinish() >> {
-            actualProperties.each {
-                output.println(it.value)
+            // important to output in the order of the enum members as parsing uses enum ordinals
+            probedSystemProperties.each {
+                def actualValue = actualProperties[it.systemPropertyKey]
```

Comment:
Is this necessary here? The default maps in Groovy are LinkedHashMaps, so they have the same ordering as you declared them.

Reply:
The problem is that they are declared sort of out of order in where these maps are actually defined for all tests.
On the other hand, the parsing relies heavily on the ordering of printed properties. I was chasing the problem for almost an hour before I realized why ordering was broken.

I think it is better to be explicit here, because the method tries to simulate the contract of the parser. It is also better, because things will not break if somebody adds a new property to the maps for JDKs later and it would happen to be out of order with the enum.

Reply:
Alright, if this makes the test less brittle then let's do it.